src/main/java/com/wavefront/proxy/ProxyForwarderImpl.java

```diff
@@ -104,6 +125,8 @@ public void forward(AppEnvelope appEnvelope) {
 
   private void send(String metricName, double metricValue, Long timestamp, String source,
                     Map<String, String> tags) {
+    // Add custom tags to the point metrics
+    tags.putAll(customTags);
```

Comment:
this is potentially problematic, imagine this is a shared map, you should use a defensive copy

Reply:
maybe just put a comment that says only new maps are supplied as the parameter tags and so no defensive copying is needed